Makefile.include

```diff
@@ -512,15 +512,23 @@ BUILDDEPS += $(RIOTBUILD_CONFIG_HEADER_C)
 BUILDDEPS += pkg-prepare
 BUILDDEPS += $(APPDEPS)
 
+# Randomize library order to protect against code re-use attacks (link time reordering)
+RIOTLINKSEED ?= -1
+ifeq ($(BUILDOSXNATIVE),1)
+  SHUFLIBS = $(shell find $(BASELIBS) -size +8c | $(RIOTTOOLS)/randsort/randsort.py --seed $(RIOTLINKSEED))
+else
+  SHUFLIBS = $(shell printf "%s\n" $(BASELIBS) | $(RIOTTOOLS)/randsort/randsort.py --seed $(RIOTLINKSEED))
+endif
+
 # Save value to verify it is not modified later
 _BASELIBS_VALUE_BEFORE_USAGE := $(BASELIBS)
 
 # Linker rule
 $(ELFFILE): FORCE
 ifeq ($(BUILDOSXNATIVE),1)
-  _LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $$(find $(BASELIBS) -size +8c) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
+  _LINK = $(if $(CPPMIX),$(LINKXX),$(LINK)) $(UNDEF) $(SHUFLIBS) $(LINKFLAGS) $(LINKFLAGPREFIX)-no_pie
```

Comment:
I don't like the use of SHUFLIBS here. It makes it look as if libraries are always shuffled.
Can we have e.g., "LINKLIBS" here, and above, use sth like:

```
if KALR:
  LINKLIBS = $(shuffle $(BASELIBS))
else:
  LINKLIBS = $(BASELIBS)
```

Reply:
Due to the OSX specific find adding another if/else is a bit annoying. Is the `find(1)` invocation on OSX really strictly necessary?

Reply:
> Is the `find(1)` invocation on OSX really strictly necessary?

I think we have to assume that. (IIRC there was an issue with empty archives...).

Maybe we can factor that ifdef out of ```ifeq ($(BUILDOSXNATIVE),1)```?

Reply:
I am ok with renaming the variable from `SHUFLIBS` to `LINKLIBS`. I don't think that it makes that much sense to add an additional disable/enable switch though. If you want a deterministic order just set `RIOTLINKSEED`.

From the `ld(1)` `--start-group` documentation:

```
The specified archives are searched repeatedly until no new undefined references are created.
```

Don't really see which benefit the "default" order provides then.